src/Components/SmartComponents/SystemsExposedTable/SystemsExposedTable.js

```diff
@@ -138,12 +138,12 @@ const SystemsExposedTable = (props) => {
                     cveList={cves}
                     updateRef={() => {
                         updateRef(items.meta, apply);
-                        fetchCveDetails(props.cve);
+                        dispatch(fetchCveDetails(props.cve));
+                        dispatch(fetchAffectedSystemsByCVE(props.cve, { ...parameters }));
```

Comment:
You don't need to dispatch `fetchAffectedSystemsByCVE`. `udpateRef` should handle it. Otherwise, you'll end up requesting affected_systems twice.

Reply:
@katerinapat `updateRef` does handle it on the other pages, but systems exposed page is special because it uses deep object filtering, so just calling `apply()` does not refresh the data because `parameters` are the same. Removing `shallowEqual` function from here would work, but that would ruin the point why deep object filtering was introduced.

```
const parameters = useSelector(
    ({ CVEDetailsPageStore }) => CVEDetailsPageStore.parameters,
    shallowEqual
);
```

So `updateRef` does not dispatch `fetchCveDetails()` nor `fetchAffectedSystemsByCVE()` by itself. So calling it manually is sufficient, I checked the network tab in debug tools. And these requests are indeed only called once.

Reply:
good observation here !!! As you said the `apply` will be triggered if the `parameters` are not the same. So in cases, they are we'll end up with 2 requests. it seems `updateRef` is not needed here.   I was able to reproduce it but github doesn't support the file format. I will upload it in the jira ticket.